### PR TITLE
GPS simulation as a compile time option.

### PR DESCRIPTION
**Pull request by bubeck**
This driver offers the functionality to simulate GPS data read from a file instead of the sensor. This allows
various flight simulations and checking, that the widgets (or other
code) behaves as expected.

To use that feature define the preprocessor constant GPS_SIMULATION
and re-compile the code. Then place a GPS file onto the SD card as
"GPS-SIM.TXT" which contains the GPS data. This file can be generated
in two different ways:
  1. Use SkyDrop and turn on "debug log" and "gps debug", record some
     movements and rename "DEBUG.LOG" into "GPS-SIM.TXT"
  2. Convert an existing IGC file into GPS, e.g.
     gpsbabel -i igc -f existing-flight.igc
              -x track,speed,course,fix=2d
              -o nmea -F GPS-SIM.TXT

### CODE REVIEW

File: skydrop/src/drivers/sensors/gps_l80.cpp [hunk 1]
```diff
@@ -119,21 +169,23 @@ void gps_parse_rmc()
...
 	longitude = loc_deg * 10000000ul + longitude;
 
+	// DEBUG("lon: loc_deg=%ld loc_min=%ld, tlen=%d\n", loc_deg, loc_min, tlen);
+
 	old_ptr = ptr;
 	ptr = find_comma(ptr);
 	tlen = ptr - old_ptr - 1;
 
-	if (tlen != 10)
+	if (tlen > 12)
```
Comment (fhorinek) [inline skydrop/src/drivers/sensors/gps_l80.cpp]: this is important check since the module always return len = 10. 
NMEA checksum is weak, we need this
please revert this change

File: skydrop/src/drivers/sensors/gps_l80.cpp [hunk 2]
```diff
@@ -90,21 +138,23 @@ void gps_parse_rmc()
...
 	latitude = loc_deg * 10000000ul + latitude;
 
+	// DEBUG("lat: loc_deg=%ld loc_min=%ld, tlen=%d\n", loc_deg, loc_min, tlen);
+
 	old_ptr = ptr;
 	ptr = find_comma(ptr);
 	tlen = ptr - old_ptr - 1;
 
-	if (tlen != 9)
+	if (tlen > 12)
```
Comment (fhorinek) [inline skydrop/src/drivers/sensors/gps_l80.cpp]: this is important check since the module always return len = 9. 
NMEA checksum is weak, we need this
please revert this change

File: skydrop/src/drivers/sensors/gps_l80.cpp (hunk 1 above)
Comment (bubeck) [reply skydrop/src/drivers/sensors/gps_l80.cpp]: I changed this, because GPS babel produces NMEA sentences which are not "our" length and I found nothing inside the specification that states, that the length is fixed. I think, it could be variabel.

I do not see a problem, because atoi_n stops when finding a "," so, there will be no overflow.

However, if you really dont want it, then we should at least use "#ifdef GPS_SIMULATION, because otherwise you will not be able to use gpsbabel to create the simulation files, which would be a pity.

File: skydrop/src/drivers/sensors/gps_l80.cpp (hunk 2 above)
Comment (bubeck) [reply skydrop/src/drivers/sensors/gps_l80.cpp]: see above

File: skydrop/src/drivers/sensors/gps_l80.cpp (hunk 1 above)
Comment (fhorinek) [reply skydrop/src/drivers/sensors/gps_l80.cpp]: I know that parser can handle the data but it caused lot of troubles since NMEA checksum is so weak that it will be correct when you miss two same characters in the sentence.
There was few random and mystic data points in log files, it took ages to debug.

Feel free to use > 12 when GPS_SIMULATION is on, it is a special case

**Comment by bubeck** [comment]
Changed back to previous behavior if reading from sensor, but relaxed for reading simulations.
